wifi-subsys/components/tools/test/test_cbor.c

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (c) 2022 Mesh4all <mesh4all.org>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @brief   Non Volatile Storage
+ *
+ * @author  eduazocar <eduardo@turpialdev.com>
+ */
+#include <stdio.h>
+#include <stdint.h>
+#include "esp_log.h"
+#include "unity.h"
+#include "serialization.h"
+
+static const char* message = "Mesh data sended";
+CborValue my_it;
+char data_received [20];
+char text[100];
+
+TEST_CASE("Encode a message", "cbor")
+{
+    esp_err_t err = encode((uint8_t* )message, &my_it);
+    if (err != ESP_OK)
+    {
+        TEST_FAIL();
+    }
+}
+
+TEST_CASE("Decode a message", "cbor")
+{
+    esp_err_t err = decode(&my_it, data_received);
+    if (err != ESP_OK)
+    {
+        TEST_FAIL();
```

Comment:
although the function respond is ESP_OK  you will be not sure that decode have working   verify that data_received will be equal  message, that will be a best test.